src/broker/broker.c

```diff
@@ -217,8 +331,20 @@ int broker_run(Broker *broker) {
                         r = MAIN_FAILED;
                 else
                         r = error_fold(r);
+
+                if (broker->do_reexec)
+                        r = MAIN_REEXEC;
+
         } while (!r);
 
+        Peer *peeri;
+        c_rbtree_for_each_entry(peeri, &broker->bus.peers.peer_tree, registry_node) {
+                socket_dispatch_write(&peeri->connection.socket);
```

Comment:
This does not reliably flush the entire queue. There are several scenarios where this will _not_ block and instead retain data in the user-space queue. In those cases, we would lose messages.

Reply:
Yes, I toke it for granted that all messages we have gotten will be sent out by `socket_dispatch_write` successfully. But if there are not many messages flying between dbus and peers, I think it's probobly fine.

By the way, do you think if we should support new client to establish connection during reexecuting? I haved checked systemd, `systemctl start/stop` doesn't work when `systemctl daemon-reexec` is running. Supporting that seems very complicated IMHO.